### PR TITLE
LIV-780: refactor "checkAndCreateRecordedEntry" function

### PR DESCRIPTION
**Pull request by Amirhalfon1**
change "checkAndCreateRecordedEntry" to "shouldCreateRecordedEntryForAppend"
split to "shouldCreateRecordedEntryForPerSession" and "shouldCreateRecordedEntryForAppend"
add "convertToApiObject" and call it 5 times (avoid code duplication)
**This PR won't change behaviour, except for:**
for append mode - add logic to check that current recorded entry (if exists) not exceeded limit (default limit is 24H)

### CODE REVIEW

File: api_v3/lib/KalturaLiveEntryService.php [hunk 1]
```diff
@@ -516,62 +518,125 @@ public function createRecordedEntryAction($entryId, $mediaServerIndex, $liveEntr
...
+	 * @throws Exception
+	 */
+	protected function shouldCreateRecordedEntryForPerSession(entry $dbRecordedEntry, LiveEntry $dbLiveEntry)
+	{
+		$recordedEntryCreationTime = $dbRecordedEntry->getCreatedAt(null);
+		$liveSessionReconnectTimeout = kConf::get('live_session_reconnect_timeout', 'local', 180);
+		$isNewSession = $dbLiveEntry->getLastBroadcastEndTime() + $liveSessionReconnectTimeout < $dbLiveEntry->getCurrentBroadcastStartTime();
+		$recordedEntryNotYetCreatedForCurrentSession = $recordedEntryCreationTime + $liveSessionReconnectTimeout < $dbLiveEntry->getCurrentBroadcastStartTime();
+		KalturaLog::debug("isNewSession [$isNewSession] getLastBroadcastEndTime [{$dbLiveEntry->getLastBroadcastEndTime()}] getCurrentBroadcastStartTime [{$dbLiveEntry->getCurrentBroadcastStartTime()}]");
+		KalturaLog::debug("recordedEntryNotYetCreatedForCurrentSession [$recordedEntryNotYetCreatedForCurrentSession] recordedEntryCreationTime [$recordedEntryCreationTime] getCurrentBroadcastStartTime [{$dbLiveEntry->getCurrentBroadcastStartTime()}]");
```
Comment (david-winder-kaltura) [inline api_v3/lib/KalturaLiveEntryService.php:610]: why we need two debug line? (can it be in one?)

File: api_v3/lib/KalturaLiveEntryService.php [hunk 2]
```diff
@@ -516,62 +518,125 @@ public function createRecordedEntryAction($entryId, $mediaServerIndex, $liveEntr
...
+	/**
+	 * Checking whether should create recorded entry for the "append" record status case
+	 * @param entry $dbRecordedEntry
+	 * @return bool
+	 * @throws Exception
+	 */
+	protected function shouldCreateRecordedEntryForAppend(entry $dbRecordedEntry)
+	{
+		$recordedEntryCreationTime = $dbRecordedEntry->getCreatedAt(null);
+		$isKalturaRecordedLive = $dbRecordedEntry->getSourceType() == EntrySourceType::KALTURA_RECORDED_LIVE; // probably can be removed
```
Comment (david-winder-kaltura) [inline api_v3/lib/KalturaLiveEntryService.php:588]: as we suspect we can remove it - lets print here something if it is false - so later we can scan the log and see if we right and it actually can be remove

Comment (Amirhalfon1) [reply api_v3/lib/KalturaLiveEntryService.php:588]: done

**Review by david-winder-kaltura — APPROVED** [review]

File: api_v3/lib/KalturaLiveEntryService.php [hunk 3]
```diff
@@ -15,8 +15,11 @@ class KalturaLiveEntryService extends KalturaEntryService
 	//amount of time for holding kLock
 	const KLOCK_CREATE_RECORDED_ENTRY_HOLD_TIMEOUT = 3;
 
+	//Default recording limit length for append recording
+	const DAY_IN_SECONDS = 86400;
+
 	//Max time from recording created time before creating new recorded entry
-	const SEVEN_DAYS_IN_SECONDS = 604800;
+	const SEVEN_DAYS_IN_SECONDS = 7 * self::DAY_IN_SECONDS;
```
Comment (MosheMaorKaltura) [inline api_v3/lib/KalturaLiveEntryService.php:22]: Use infra/general/timeOffset/kTimeConst.php

Comment (Amirhalfon1) [reply api_v3/lib/KalturaLiveEntryService.php]: done

**Pull request merged**